src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -472,6 +472,11 @@ static Map.Entry<SignatureScheme, Signature> getSignerOfPreferableAlgorithm(
             ProtocolVersion version) {
 
         PrivateKey signingKey = x509Possession.popPrivateKey;
+
+        ECParameterSpec params = x509Possession.getECParameterSpec();
+        NamedGroup namedGroup = params != null
+                ? NamedGroup.valueOf(params) : null;
+
```

Comment:
X509Possession.java might be a better place for the update, by caching  the named group of the private key.